File: PiDEV-JAVA-Integrated-Final-v8/src/main/java/models/PrescriptionMedicale.java
package models;

import java.util.Date;

public class PrescriptionMedicale {
    private int id;
    private Date datePrescription;
    private String medicaments;
    private String recommandations;
    private String suivi;
    private int suiviTherapeutiqueId;

    public PrescriptionMedicale() {}

    public PrescriptionMedicale(int id, Date datePrescription, String medicaments,
                                  String recommandations, String suivi, int suiviTherapeutiqueId) {
        this.id = id;
        this.datePrescription = datePrescription;
        this.medicaments = medicaments;
        this.recommandations = recommandations;
        this.suivi = suivi;
        this.suiviTherapeutiqueId = suiviTherapeutiqueId;
    }

    public int getId() { return id; }
    public void setId(int id) { this.id = id; }

    public Date getDatePrescription() { return datePrescription; }
    public void setDatePrescription(Date datePrescription) { this.datePrescription = datePrescription; }

    public String getMedicaments() { return medicaments; }
    public void setMedicaments(String medicaments) { this.medicaments = medicaments; }

    public String getRecommandations() { return recommandations; }
    public void setRecommandations(String recommandations) { this.recommandations = recommandations; }

    public String getSuivi() { return suivi; }
    public void setSuivi(String suivi) { this.suivi = suivi; }

    public int getSuiviTherapeutiqueId() { return suiviTherapeutiqueId; }
    public void setSuiviTherapeutiqueId(int suiviTherapeutiqueId) { this.suiviTherapeutiqueId = suiviTherapeutiqueId; }

    @Override
    public String toString() { return medicaments; }
}

File: symfony-integ/src/Entity/PrescriptionMedicale.php
<?php
namespace App\Entity;
use App\Repository\PrescriptionMedicaleRepository;
use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity(repositoryClass: PrescriptionMedicaleRepository::class)]
#[ORM\Table(name: 'prescription_medicale')]
class PrescriptionMedicale
{
    #[ORM\Id] #[ORM\GeneratedValue] #[ORM\Column]
    private ?int $id = null;
    #[ORM\Column(name: 'date_prescription', type: 'datetime')] private ?\DateTimeInterface $datePrescription = null;
    #[ORM\Column(length: 255)]                                  private ?string $medicaments = null;
    #[ORM\Column(length: 255, nullable: true)]                  private ?string $recommandations = null;
    #[ORM\Column(length: 255, nullable: true)]                  private ?string $suivi = null;
    #[ORM\Column(name: 'suivi_therapeutique_id')]               private ?int $suiviTherapeutiqueId = null;

    public function getId(): ?int { return $this->id; }
    public function getDatePrescription(): ?\DateTimeInterface { return $this->datePrescription; }
    public function setDatePrescription(\DateTimeInterface $v): static { $this->datePrescription = $v; return $this; }
    public function getMedicaments(): ?string { return $this->medicaments; }
    public function setMedicaments(string $v): static { $this->medicaments = $v; return $this; }
    public function getRecommandations(): ?string { return $this->recommandations; }
    public function setRecommandations(?string $v): static { $this->recommandations = $v; return $this; }
    public function getSuivi(): ?string { return $this->suivi; }
    public function setSuivi(?string $v): static { $this->suivi = $v; return $this; }
    public function getSuiviTherapeutiqueId(): ?int { return $this->suiviTherapeutiqueId; }
    public function setSuiviTherapeutiqueId(int $v): static { $this->suiviTherapeutiqueId = $v; return $this; }
}
